packages/runtime/src/widgets/Form/Radio.tsx

```diff
@@ -21,15 +33,42 @@ export type RadioWidgetProps = FormInputProps<string> &
     onChange?: EnsembleAction;
   };
 
+interface RadioOptionsProps {
+  disabled: boolean;
+  value: string | number;
+  children: React.ReactElement | string;
+  style: EnsembleWidgetStyles;
+}
+
 export const RadioWidget: React.FC<RadioWidgetProps> = (props) => {
+  const { "item-template": itemTemplate, onChange, ...rest } = props;
   const [value, setValue] = useState<string | number | undefined>(undefined);
   const { values, rootRef } = useRegisterBindings(
-    { ...props, initialValue: props.value, value, widgetName },
-    props.id,
+    { ...rest, initialValue: rest.value, value, widgetName },
+    rest.id,
     { setValue },
   );
-  const action = useEnsembleAction(props.onChange);
 
+  // update initial value
+  useEffect(() => {
+    setValue(values?.initialValue?.toString());
+  }, [values?.initialValue]);
+
+  // update form field initial value
+  const formInstance = Form.useFormInstance();
+
+  useEffect(() => {
+    if (formInstance) {
+      formInstance.setFieldsValue({
+        [values?.id ?? values?.label ?? ""]: value,
+      });
+    }
+  }, [value, formInstance]);
```

Comment:
Why is this needed?

Reply:
@evshi this is needed to set the initial values of radio widget in the form instance, so it will validate that values or we can get those values on form submit time.

we did this in all form elements like, dropdown, radio, checkbox, textInput, Date and so on